tw-tasks-core/src/main/java/com/transferwise/tasks/processing/TasksProcessingService.java

```diff
@@ -212,9 +212,10 @@ protected ProcessTasksResponse processTasks(GlobalProcessingState.Bucket bucket)
         mdcService.with(() -> {
           mdcService.put(task);
           try {
-            ProcessTaskResponse processTaskResponse = grabTaskForProcessing(bucketId, task);
 
-            coreMetricsTemplate.registerTaskGrabbingResponse(bucketId, type, priority, processTaskResponse);
+            long epochMilliBeforeProcessing = System.currentTimeMillis();
```

Comment:
Record time before `grabTaskForProcessing` to remove the task processing time from metric.

Maybe would be cleaner to add `epochMilliBeforeProcessing` to `ProcessTaskResponse` and have it be returned from `grabTaskForProcessing` 🤔

Reply:
agree with proposal, could be part of the response

Reply:
So it turns out that in the successful flow (code = `OK`) the final task grabbing / execution is handed off to another thread which is not awaited so can just use the time the metric method is called at 😄 

https://github.com/transferwise/tw-tasks-executor/blob/f1b9e2d1c47cd5148a3a66d64536dfb76d61bc85/tw-tasks-core/src/main/java/com/transferwise/tasks/processing/TasksProcessingService.java#L339-L345